### ISSUE TITLE
[HOLD for payment][$2000] "report no longer exists" message appears when try to edit message

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#19707]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. go to any chat
2. send any message
3. Hover over your sent message and click "Reply in Thread" 
4. Go to thread
5. Click your profile icon in the new chat thread > click "Leave thread"
6. Find previous chat 
7. Hover over the last sent message and click the edit pencil
8. Type an edit to your previously sent message > enter
9. Get a "Report no longer exists" error

## Expected Result: 
user should be able to edit the message

## Actual Result: 
report no longer exists message appears when try to edit message

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [X] Android / native
- [X] Android / Chrome
- [X] iOS / native
- [X] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.19-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/cb67a0e5-eebb-4864-9824-ba7064b7b801

https://github.com/Expensify/App/assets/43996225/5b6cb099-265f-4470-a6f5-0622d4a01b5d

**Expensify/Expensify Issue URL:**
**Issue reported by:** @gadhiyamanan 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1684936594996169

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d37108580b2e2ecb</li>
        <li>Upwork Job ID: 1692659620193611776</li>
        <li>Last Price Increase: 2023-09-07</li>
    </ul>
</details>

**Comment by Christinadobrzyn** [comment]
I can reproduce this but I think it might be the same fix as this - https://github.com/Expensify/App/issues/19559#issuecomment-1568708454 asking

**Comment by Christinadobrzyn** [comment]
Great questions - @c3024! 

> What does Leave Thread do exactly?

My understanding is it allows a user to 'unsubscribe' from a chat thread (generally a group thread) where they no longer need to participate. V2 should have a mute option 

>  Is it same as clicking on Conceirge chat with the thread continuing to exist or does it mean deleting the thread altogether?

So the thread will exist, you just can choose to exit it to no longer be part of it

cc @stitesExpensify 

I've added a Hold label as we'll wait to see if this root issue is similar to https://github.com/Expensify/App/issues/19559#issuecomment-1568989975

**Comment by Christinadobrzyn** [comment]
Still on hold

**Comment by Christinadobrzyn** [comment]
Still on hold for - https://github.com/Expensify/App/issues/19559#issuecomment-1568989975

**Comment by Christinadobrzyn** [comment]
Still on hold for - https://github.com/Expensify/App/issues/19559#issuecomment-1568989975

**Comment by Christinadobrzyn** [comment]
Still on hold for - https://github.com/Expensify/App/issues/19559#issuecomment-1568989975

**Comment by Christinadobrzyn** [comment]
hold for - https://github.com/Expensify/App/issues/19559#issuecomment-1568989975

**Comment by Christinadobrzyn** [comment]
I'm going to be ooo until August 31st so going to unassign and assign a new teammate. 

At this time, we're on hold for https://github.com/Expensify/App/issues/19559#issuecomment-1568989975

**Comment by johncschuster** [comment]
Still on hold for https://github.com/Expensify/App/issues/19559#issuecomment-1568989975

**Comment by johncschuster** [comment]
Still on hold. It looks like the linked issue is almost complete.

**Comment by johncschuster** [comment]
The engineer working on the issue above was OOO last week. Looks like they're back to working on it now

**Comment by johncschuster** [comment]
Asked for an update on [#19559 ](https://github.com/Expensify/App/issues/19559)

**Comment by Christinadobrzyn** [comment]
I'm back from ooo - thanks for handling this @johncschuster I can take it off your plate - thanks!

**Comment by johncschuster** [comment]
Thanks, @Christinadobrzyn!

**Comment by Christinadobrzyn** [comment]
I think this can be moved to weekly while waiting on https://github.com/Expensify/App/issues/19559#issuecomment-1568989975

**Comment by Christinadobrzyn** [comment]
waiting on - https://github.com/Expensify/App/issues/19559#issuecomment-1568989975

**Comment by Christinadobrzyn** [comment]
waiting on - https://github.com/Expensify/App/issues/19559#issuecomment-1568989975

**Comment by Christinadobrzyn** [comment]
This is closed - https://github.com/Expensify/App/issues/19078

Testing this again... I can still reproduce the error in the OP so I think this still needs to be worked on. Adding external label

**Comment by Christinadobrzyn** [comment]
waiting on proposals ... or let me know if you think we should do something else @aimane-chnaif

**Comment by b4s36t4** [comment]
I think is from backend. Whenever we're creating a thread we're adding a new property `childReportID` in which case of every thread we create should be containing this.

But when we leave the thread this `childReportID` is not removing and maybe because of this backend is giving error.

<img width="353" alt="Screenshot 2023-08-25 at 3 53 31 AM" src="https://github.com/Expensify/App/assets/59088937/740c8ced-0a38-4c38-b68f-2afe1a07c9c9">

In the image the first reportAction is a message which was existed from thread, 2nd one is direct message no thread nothing.

**Comment by Christinadobrzyn** [comment]
Hi @aimane-chnaif can you review this proposal - https://github.com/Expensify/App/issues/19707#issuecomment-1692492641 so we can keep this moving forward? thank you!

**Comment by Christinadobrzyn** [comment]
just a nudge for @aimane-chnaif on ^ https://github.com/Expensify/App/issues/19707#issuecomment-1693694523

**Comment by b4s36t4** [comment]
<img width="407" alt="Screenshot 2023-08-29 at 2 18 49 PM" src="https://github.com/Expensify/App/assets/59088937/6d176fb8-6c20-4fb6-91f4-c734e994d663">


@aimane-chnaif No even after refresh it's giving same error but we're not showing that's all.

**Comment by ShogunFire** [comment]
@aimane-chnaif I also believe this is backend, the only reason it works after refresh is because we call openReport where the message is right. After refresh if we retry to edit the comment the api still returns an error in UpdateComment, we just don't show the red dot and the error again for some reason

**Comment by ShogunFire** [comment]
I also believe this is back end, the message is right after refresh because the message is good in openreport but updatecomment still returns an error

**Comment by Christinadobrzyn** [comment]
@aimane-chnaif do you think we should change this to an `Internal` issue?

**Comment by b4s36t4** [comment]
@Christinadobrzyn I think you should, because I see no error from frontend is causing this issue.

**Comment by Christinadobrzyn** [comment]
Making this `Internal` and adding @tylerkaraszewski to check on https://github.com/Expensify/App/issues/19707#issuecomment-1710801634 for us

**Comment by tylerkaraszewski** [comment]
I need to catch up on this. It seems the issue is in the `UpdateComment` API.

**Comment by tylerkaraszewski** [comment]
So the issue here is that we create a thread (which is a report), and then we leave the thread, but we are the owner of the parent reportAction of this report. So when we go to edit the parent comment, we attempt to notify the participants that the comment has been edited, but we cannot get the list of participants, because we've unshared the thread with ourself.

`OpenReport` seems to work to open a thread you've previously left. I'm not sure how this works. It seems like it wouldn't, but it also seems like it should. I haven't investigated in depth so far.

Here are some options for fixing this:

1. Disallow leaving a thread you created. This might resolve the above weirdness with `OpenReport`, even though that seems to work.
2. Don't notify participants that the comment has changed if the owner of that comment has left the thread.
3. Make the backend do the notification regardless of if the user has access to the thread. This probably requires a lot of work because I think it breaks existing permission models.

**Comment by Christinadobrzyn** [comment]
@aimane-chnaif do you have an opinion on these options? https://github.com/Expensify/App/issues/19707#issuecomment-1724537871

**Comment by b4s36t4** [comment]
I feel preventing user from leaving thread if he is the owner of the report make sense. My vote for option 1

**Comment by cristipaval** [comment]
@tylerkaraszewski I have 2 internal PRs for [this issue](https://github.com/Expensify/App/issues/19559) which might have the same root cause as this one.

**Comment by cristipaval** [comment]
> @tylerkaraszewski I have 2 internal PRs for [this issue](https://github.com/Expensify/App/issues/19559) which might have the same root cause as this one.

Ok, I posted this last night after a quick overview of the issue before going to sleep. Here are more details @tylerkaraszewski 

- it seems to me that this issue has the exact same root cause as [my issue](https://github.com/Expensify/App/issues/19559)
- [Here](https://github.com/Expensify/App/issues/19559#issuecomment-1593769635) is the internal root cause
- [Here](https://github.com/Expensify/App/issues/19559#issuecomment-1594304496) is the solution that we agreed that we should implement to fix the issue
- Auth PR [here](https://github.com/Expensify/Auth/pull/8145) and Web-E PR [here](https://github.com/Expensify/Web-Expensify/pull/37878). Would love if you could review!

**Comment by Christinadobrzyn** [comment]
Sounds like we're determining the best next steps and a review of [this](https://github.com/Expensify/App/issues/19707#issuecomment-1730198490) by @tylerkaraszewski!

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by Christinadobrzyn** [comment]
keeping a watch on KI retests!

**Comment by cristipaval** [comment]
> Issue not reproducible during KI retests. (First week)

Interesting. 
A fix was merged and will hit staging with the next Web-E deployment.

**Comment by Christinadobrzyn** [comment]
If this is not reproducible in KI retests, I'm assuming we'll still pay this out? @cristipaval 

seems like we'll be paying @aimane-chnaif $1000 (based on the old payment structure).

**Comment by cristipaval** [comment]
@Christinadobrzyn The fix is now in production, and this issue could be closed after the payments. The fix was implemented completely internally, in Auth and Web-E. No code review was done by any C+. In the [other issue](https://github.com/Expensify/App/issues/19559) that had the same root cause, we're paying the reporting bonus only.

**Comment by Christinadobrzyn** [comment]
Thanks @cristipaval it looks like this reporter is different from the one in https://github.com/Expensify/App/issues/19559

So I could add this reporter to that other issue to ensure they are paid, or I could just pay them here. I think it might be better to combine the reporter payments into one GH. 

If that's the case then there would be no payments issued through this Upwork job. What do you think?

**Comment by cristipaval** [comment]
> Thanks @cristipaval it looks like this reporter is different from the one in #19559
> 
> So I could add this reporter to that other issue to ensure they are paid, or I could just pay them here. I think it might be better to combine the reporter payments into one GH.
> 
> If that's the case then there would be no payments issued through this Upwork job. What do you think?

I don't know how the payment process works. You can pay them the easier it is for you.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by Christinadobrzyn** [comment]
I'm going to pay @gadhiyamanan for reporting this in this GH. 

@gadhiyamanan can you accept my job here - https://www.upwork.com/jobs/~01ea964ec786dd78e1

**Comment by gadhiyamanan** [comment]
@Christinadobrzyn offer accepted, thanks

**Comment by cristipaval** [comment]
What's the payment status here? Can we close the issue?

**Comment by Christinadobrzyn** [comment]
Sorry for the delay- paid out based on https://github.com/Expensify/App/issues/19707#issuecomment-1745799697
